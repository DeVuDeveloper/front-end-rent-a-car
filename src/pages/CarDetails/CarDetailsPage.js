import React from 'react';
import CarDetails from './CarDetails';
import Sidebar from '../../components/navigation/Sidebar';
import Hamburger from '../../components/navigation/Hamburger';
import style from './Detail.module.scss';

const CarDetailsPage = () => (
  <section className={style.car_details_page}>
    <div className={style.header}>
      <Hamburger />
    </div>
    <div className={style.sidebar}>
      <Sidebar />
    </div>
    <CarDetails />
  </section>
);

export default CarDetailsPage;
