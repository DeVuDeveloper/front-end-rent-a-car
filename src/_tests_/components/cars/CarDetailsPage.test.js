import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import CarDetailsPage from '../../../pages/CarDetails/CarDetailsPage';
import store from '../../../redux/configureStore';

it('should render CarDetailsPage component correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <CarDetailsPage />
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
