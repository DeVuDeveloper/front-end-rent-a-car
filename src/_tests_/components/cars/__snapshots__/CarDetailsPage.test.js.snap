// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CarDetailsPage component correctly 1`] = `
<section
  className="car_details_page"
>
  <div
    className="header"
  >
    <header
      className="flex flex-col px-8  h-fit fixed top-0 w-screen shadow-xl md:hidden"
    >
      <a
        className="text-lg text-slate-50"
        href="/"
      >
        <img
          alt="logo"
          className="w-24 block"
          src="logo.png"
        />
      </a>
      <nav>
        <div
          className="text-slate-50 mobile-btns"
        >
          <button
            className=" text-lime-500 block"
            onClick={[Function]}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="h-8"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
                fillRule="evenodd"
              />
            </svg>
          </button>
          <button
            className="hidden text-slate-900"
            onClick={[Function]}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="h-8"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
                fillRule="evenodd"
              />
            </svg>
          </button>
        </div>
        <div
          className="hidden"
        >
          <ul
            className="flex flex-col text-slate-900 font-bold text-lg gap-8 items-center mt-44"
          >
            <li>
              <a
                href="/home"
              >
                Cars
              </a>
            </li>
            <li>
              <a
                href="/reserve"
              >
                Reserve
              </a>
            </li>
            <li>
              <a
                href="/reservations"
              >
                My reservations
              </a>
            </li>
            <li>
              <a
                href="/"
              >
                Add car
              </a>
            </li>
            <li>
              <a
                href="/delete"
              >
                Delete car
              </a>
            </li>
          </ul>
        </div>
      </nav>
      <div
        className="hidden flex flex-col items-center mt-32"
      >
        <div
          className="flex gap-2"
        >
          <a
            href="/"
          >
            <img
              alt="social-icon"
              className="w-8"
              src="twitter-icon.png"
            />
          </a>
          <a
            href="/"
          >
            <img
              alt="social-icon"
              className="w-8"
              src="facebook-icon.png"
            />
          </a>
          <a
            href="/"
          >
            <img
              alt="social-icon"
              className="w-8"
              src="linkedin-icon.png"
            />
          </a>
          <a
            href="/"
          >
            <img
              alt="social-icon"
              className="w-8"
              src="github-icon.png"
            />
          </a>
        </div>
        <p
          className="text-gray-700 font-bold mt-2"
        >
          © 2022
        </p>
      </div>
    </header>
  </div>
  <div
    className="sidebar"
  >
    <div
      className="flex flex-col w-64 sidebar-wrapper"
    >
      <span
        className="nav-header"
      >
        <h2>
          Rent a Car
        </h2>
      </span>
      <span
        className="user-mail"
      />
      <div
        className="flex-1 flex flex-col pt-3 pb-4 overflow-y-auto"
      >
        <div
          className="flex items-center flex-shrink-0 px-4"
        >
          <a
            href="/"
          >
            <img
              alt="logo"
              className="w-40"
              src="logo.png"
            />
          </a>
        </div>
        <nav
          aria-label="Sidebar"
          className="mt-12 flex-1 desktop-nav"
        >
          <div
            className="pl-3 uppercase font-black text-md"
          >
            <a
              aria-current="page"
              className="bg-lime-500 text-slate-50 group flex items-center pl-5 py-3 hover:text-slate-900"
              href="/home"
            >
              Cars
            </a>
            <a
              className="text-slate-900 hover:bg-lime-200 group flex items-center pl-5 py-3 hover:text-slate-900"
              href="reserve"
            >
              Reserve
            </a>
            <a
              className="text-slate-900 hover:bg-lime-200 group flex items-center pl-5 py-3 hover:text-slate-900"
              href="/reservations"
            >
              My reservations
            </a>
            <a
              className="text-slate-900 hover:bg-lime-200 group flex items-center pl-5 py-3 hover:text-slate-900"
              href="#"
            >
              Add car
            </a>
            <a
              className="text-slate-900 hover:bg-lime-200 group flex items-center pl-5 py-3 hover:text-slate-900"
              href="delete"
            >
              Delete car
            </a>
            <span
              className="logout"
            >
              <button
                className="logout"
                onClick={[Function]}
                type="button"
              >
                <h4>
                  LOGOUT
                </h4>
              </button>
            </span>
          </div>
        </nav>
      </div>
      <div
        className="flex-shrink-0 flex-200 p-4"
      >
        <div
          className="flex gap-2"
        >
          <a
            href="/"
          >
            <img
              alt="social-icon"
              className="w-6"
              src="twitter-icon.png"
            />
          </a>
          <a
            href="/"
          >
            <img
              alt="social-icon"
              className="w-6"
              src="facebook-icon.png"
            />
          </a>
          <a
            href="/"
          >
            <img
              alt="social-icon"
              className="w-6"
              src="linkedin-icon.png"
            />
          </a>
          <a
            href="/"
          >
            <img
              alt="social-icon"
              className="w-6"
              src="github-icon.png"
            />
          </a>
        </div>
        <p
          className="text-gray-700 font-bold mt-2"
        >
          © 2022
        </p>
      </div>
    </div>
  </div>
  <section
    className="show_container"
  >
    <div
      className="show_img"
    >
      <div
        className="img_div"
      >
        <img
          alt="Car"
        />
      </div>
    </div>
    <div
      className="show_content"
    >
      <div
        className="show_header"
      >
        <h2 />
        <p>
          -$200 deposit upon any Rental Purchase!
        </p>
      </div>
      <div
        className="show_details"
      >
        <div
          className="car_details"
        >
          <p
            className="details"
          />
        </div>
        <div
          className="car_detail"
        >
          <p
            className="details"
          >
            Rent daily cost
          </p>
          <p
            className="details"
          >
            $
          </p>
        </div>
        <div
          className="car_details"
        >
          <p
            className="details"
          >
            Car type
          </p>
          <p
            className="details"
          />
        </div>
        <div
          className="car_detail"
        >
          <p
            className="details"
          >
            Transmission
          </p>
          <p
            className="details"
          />
        </div>
        <div
          className="show_discount"
        >
          <p>
            <span
              className="dis_amt"
            >
              NaN
              % APR
            </span>
            <span
              className="dis_text"
            >
              representative
            </span>
          </p>
        </div>
      </div>
      <div
        className="more_cars"
      >
        <a
          aria-current={null}
          className="d_flex"
          href="/home"
          onClick={[Function]}
        >
          <span
            className="discover"
          >
            DISCOVER MORE CARS
          </span>
          <svg
            className="success"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
                "fontSize": 40,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M16 12l-6 6V6z"
              />
            </g>
          </svg>
        </a>
      </div>
      <a
        aria-current={null}
        className="btn_container"
        href="/add_reservations/undefined"
        onClick={[Function]}
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          <span>
            Reserve
          </span>
           
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
                "fontSize": 30,
              }
            }
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M10.293 15.707a1 1 0 010-1.414L14.586 10l-4.293-4.293a1 1 0 111.414-1.414l5 5a1 1 0 010 1.414l-5 5a1 1 0 01-1.414 0z"
              fillRule="evenodd"
            />
            <path
              clipRule="evenodd"
              d="M4.293 15.707a1 1 0 010-1.414L8.586 10 4.293 5.707a1 1 0 011.414-1.414l5 5a1 1 0 010 1.414l-5 5a1 1 0 01-1.414 0z"
              fillRule="evenodd"
            />
          </svg>
        </button>
      </a>
    </div>
  </section>
</section>
`;
